plugins/scaffolder-backend/src/scaffolder/tasks/TaskWorker.ts

```diff
@@ -83,14 +103,16 @@ export class TaskWorker {
     return new TaskWorker({
       taskBroker: taskBroker,
       runners: { workflowRunner },
+      concurrentTasksLimit,
     });
   }
 
   start() {
     (async () => {
       for (;;) {
+        await this.taskQueue.onEmpty();
```

Comment:
I'm assuming onEmpty waits for the queue to be completely empty? In that case we'll only process one task at a time. I think to jhaal's point it's best to use a tdd approach here and set up unit tests for the behavior that we want first.

Reply:
@Rugvip onEmpty awaits until the queue is empty, ie. no tasks are waiting to be started. It is not waiting for all tasks are done. I have been validating the funcationality with the long-running action while developing the feature, and I know `concurrentTasksLimit` is working as expected. But I will include tests to demostrate the functionality this week.

Reply:
Ah alright, makes sense but not what I would've expected that to do xP

This'll still pick up tasks before we're actually ready to process them though right?

Reply:
Hi @Rugvip, it will pick up **one** task when the waiting queue is empty.

Reply:
I added tests to show the concurrent tasks on a single worker.